nilearn/plotting/tests/test_displays.py

```diff
@@ -53,6 +56,58 @@ def test_tiled_slicer():
     slicer.close()
 
 
+def test_mosaic_slicer():
+    img = load_mni152_template()
+    # default cut_coords=None
+    slicer = MosaicSlicer.init_with_figure(img=img)
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+    # Forcing a layout here, to test the locator code
+    with tempfile.TemporaryFile() as fp:
+        slicer.savefig(fp)
+
+    # cut_coords as an integer
+    slicer = MosaicSlicer.init_with_figure(img=img, cut_coords=4)
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+    # cut_coords as a tuple
+    slicer = MosaicSlicer.init_with_figure(img=img, cut_coords=(4, 5, 2))
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+    # test title
+    slicer.title('Showing mosaic mode')
+
+    # test when img is None or False while initializing figure
+    slicer = MosaicSlicer.init_with_figure(img=None, cut_coords=None)
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+    # same test but cut_coords as integer and tuple
+    slicer = MosaicSlicer.init_with_figure(img=None, cut_coords=5)
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+    slicer = MosaicSlicer.init_with_figure(img=None, cut_coords=(1, 1, 1))
+    slicer.add_overlay(img, cmap=plt.cm.gray, colorbar=True)
+
+    # assert raises ValueError
+    pytest.raises(ValueError, MosaicSlicer.init_with_figure,
+                  img=None, cut_coords=(5, 4))
+
+    slicer.close()
+
+
+def test_demo_mosaic_slicer():
+    # cut_coords as tuple of length 3
+    mslicer = MosaicSlicer(cut_coords=(1, 1, 1))
+    img = load_mni152_template()
+    mslicer.add_overlay(img, cmap=plt.cm.gray)
+    # cut_coords as integer
+    mslicer = MosaicSlicer(cut_coords=5)
+    mslicer.add_overlay(img, cmap=plt.cm.gray)
+    # cut_coords as dictionary
+    mslicer = MosaicSlicer(cut_coords={'x': [10, 20],
+                                       'y': [30, 40],
+                                       'z': [15, 16]})
```

Comment:
Here we could check than `mslicer` has the provided cut coordinates in the same way.